### PR TITLE
fix: issue-1593- validate access token on login register

### LINKED ISSUE #1593: [BUG]: Validate AccessToken on login/register
## Description

AD access token is not being validated, and it should be to prevent the use of forged tokens

### Steps to reproduce

Try to login with a forged access token.

### Expected result

User should not be able to login or register with an invalid access token

### Actual result

User might login with a forged access token.

### PR DESCRIPTION
**Pull request by joaofrparreira**
Fixes:
- #1593 

## Proposed Changes

  - Validate AD access token on user login/register

This pull request closes #1593

### CODE REVIEW

File: backend/src/modules/azure/applications/register-or-login.azure.use-case.ts [hunk 1]
```diff
@@ -75,7 +100,7 @@ export class RegisterOrLoginAzureUseCase implements RegisterOrLoginAzureUseCaseI
 			userToAuthenticate.avatar = avatarUrl;
 		}
 
-		return signIn(userToAuthenticate, this.getTokenService, 'azure');
+		return await signIn(userToAuthenticate, this.getTokenService, 'azure');
```
Comment (nunocaseiro) [inline backend/src/modules/azure/applications/register-or-login.azure.use-case.ts:103]:
```suggestion
		return signIn(userToAuthenticate, this.getTokenService, 'azure');
```
I think you don't need the `await` here

File: backend/src/modules/azure/services/auth.azure.service.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
 
 export type AzureUserFound = {
 	id: string;
 	mail: string;
 	displayName: string;
 	userPrincipalName: string;
 	createdDateTime: Date;
+	accountEnabled: boolean;
+	deletedDateTime: Date | null;
 };
```
Comment (nunocaseiro) [inline backend/src/modules/azure/services/auth.azure.service.ts:17]: I know this was already here but what do you think about moving this to a "types" folder? Does it make sense to you?

File: backend/src/modules/azure/services/auth.azure.service.ts [hunk 3]
```diff
@@ -68,4 +79,37 @@ export default class AuthAzureService implements AuthAzureServiceInterface {
 	fetchUserPhoto(userId: string) {
 		return this.graphClient.api(`/users/${userId}/photo/$value`).get();
 	}
+
+	async getADUsers(): Promise<Array<AzureUserDTO>> {
```
Comment (nunocaseiro) [inline backend/src/modules/azure/services/auth.azure.service.ts:83]: Is this method being used or are you just preparing the code to the next feature?

**Review by nunocaseiro — CHANGES_REQUESTED** [review]

File: backend/src/modules/azure/services/auth.azure.service.ts (hunk 2 above)
Comment (joaofrparreira) [reply backend/src/modules/azure/services/auth.azure.service.ts:17]: makes all the sense.

**Pull request merged**